site/src/pages/RoadmapPage/sidebar/ProgramRequirementsList.tsx

```diff
@@ -30,13 +30,28 @@ import { useClearedCourses } from '../../../hooks/planner';
 import { useTransferredCredits } from '../../../hooks/transferCredits';
 import { useIsLoggedIn } from '../../../hooks/isLoggedIn';
 
+interface CompletedCourseSetWithType {
+  [k: string]: {
+    units: number;
+    transferType: 'AP' | 'Course';
```

Comment:
Make this property optional instead of drilling two props all the way down. Then, for non-transferred courses, simply leave it as undefined

Reply:
Additionally, instead of repeating `'AP' | 'Course'`, you should read the type directly from `TransferredCourseWithType` after exporting:
```suggestion
    transferType?: TransferredCourseWithType['transferType'];
```

Reply:
Instead of making `transferType` within `TransferredCourseWithType` optional and then exporting it, can I just add `transferType` as an optional property of `TransferredCourse`, and then just use `TransferredCourse` instead of `TransferredCourseWithType`?

Reply:
I don't necessarily think this is the correct approach for one main reason: `TransferredCourse` is the interface that's used to store items in the slice, and the slice is ONLY for explicitly transferred courses, not rewarded courses from AP. We should not allow the ambiguity that maybe AP rewards should be allowed to be stored in the `transferCreditsSlice.transferredCourses: TransferredCourse[]` array, meaning that `TransferredCourse` should not allow that property.

Reply:
This is something we should do when consolidating types though when we refactor

Reply:
Reverted back to using `TransferredCourseWithType`, although still using the modified `CompletedCourseSet`.